py/desispec/validredshifts.py

```diff
@@ -0,0 +1,152 @@
+
+    ############################ Load data ############################
+
+    columns_redshifts = ['TARGETID', 'CHI2', 'Z', 'ZERR', 'ZWARN', 'SPECTYPE', 'DELTACHI2']
+    columns_fibermap = ['TARGETID', 'COADD_FIBERSTATUS', 'TARGET_RA', 'TARGET_DEC', 'DESI_TARGET', 'BGS_TARGET']
+    columns_emline = ['TARGETID', 'OII_FLUX', 'OII_FLUX_IVAR']
+    columns_qso_mgii = ['TARGETID', 'IS_QSO_MGII']
+    columns_qso_qn = ['TARGETID', 'Z_NEW', 'ZERR_NEW', 'IS_QSO_QN_NEW_RR', 'C_LYA', 'C_CIV', 'C_CIII', 'C_MgII', 'C_Hbeta', 'C_Halpha']
+
+    dir_path = os.path.dirname(redrock_path)
+    qso_mgii_path = os.path.join(dir_path, os.path.basename(redrock_path).replace('redrock-', 'qso_mgii-'))
+    qso_qn_path = os.path.join(dir_path, os.path.basename(redrock_path).replace('redrock-', 'qso_qn-'))
+    if emline_path is None:
+        emline_path = os.path.join(dir_path, os.path.basename(redrock_path).replace('redrock-', 'emline-'))
+
+    tmp_redshifts = Table(fitsio.read(redrock_path, ext='REDSHIFTS', columns=columns_redshifts))
+    tmp_fibermap = Table(fitsio.read(redrock_path, ext='FIBERMAP', columns=columns_fibermap))
+
+    tmp_qso_mgii = Table(fitsio.read(qso_mgii_path, columns=(columns_qso_mgii)))
+    tmp_qso_qn = Table(fitsio.read(qso_qn_path, columns=(columns_qso_qn)))
+    if not ignore_emline:
+        tmp_emline = Table(fitsio.read(emline_path, columns=(columns_emline)))
+
+    # Sanity check
+    tid = tmp_redshifts['TARGETID'].copy()
+    if not (np.all(tid==tmp_fibermap['TARGETID']) \
+            # and np.all(tid==tmp_emline['TARGETID']) \
+            and np.all(tid==tmp_qso_mgii['TARGETID']) \
+            and np.all(tid==tmp_qso_qn['TARGETID'])):
+        raise ValueError
+    if (not ignore_emline) and (not np.all(tid==tmp_emline['TARGETID'])):
+        raise ValueError
+
+    tmp_fibermap.remove_column('TARGETID')
+    tmp_qso_mgii.remove_column('TARGETID')
+    tmp_qso_qn.remove_column('TARGETID')
+    if not ignore_emline:
+        tmp_emline.remove_column('TARGETID')
+
+    if not ignore_emline:
+        cat = hstack([tmp_redshifts, tmp_fibermap, tmp_qso_mgii, tmp_qso_qn, tmp_emline], join_type='inner')
+    else:
+        cat = hstack([tmp_redshifts, tmp_fibermap, tmp_qso_mgii, tmp_qso_qn], join_type='inner')
+
+    if return_target_columns:
+        # Bitmask definitions: https://github.com/desihub/desitarget/blob/master/py/desitarget/data/targetmask.yaml
+        cat['LRG'] = cat['DESI_TARGET'] & 2**0 > 0
+        cat['ELG'] = cat['DESI_TARGET'] & 2**1 > 0
+        cat['QSO'] = cat['DESI_TARGET'] & 2**2 > 0
+        cat['ELG_LOP'] = cat['DESI_TARGET'] & 2**5 > 0
+        cat['ELG_HIP'] = cat['DESI_TARGET'] & 2**6 > 0
+        cat['ELG_VLO'] = cat['DESI_TARGET'] & 2**7 > 0
+        cat['BGS_ANY'] = cat['DESI_TARGET'] & 2**60 > 0
+        cat['BGS_FAINT'] = cat['BGS_TARGET'] & 2**0 > 0
+        cat['BGS_BRIGHT'] = cat['BGS_TARGET'] & 2**1 > 0
```

Comment:
Please use `desitarget.targetmask.desi_mask` and `bgs_mask` here instead of hardcoded numbers.